packages/labs/cli/src/lib/generate/generate.ts

```diff
@@ -0,0 +1,36 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import {Analyzer} from '@lit-labs/analyzer';
+import {AbsolutePath} from '@lit-labs/analyzer/lib/paths.js';
+
+const frameworkGenerators = {
+  react: () => import('../generate/react.js'),
+};
+
+type FrameworkName = keyof typeof frameworkGenerators;
+
+export const run = async (
+  {packageRoot, frameworks}: {packageRoot: string; frameworks: string[]},
+  console: Console
+) => {
+  const analyzer = new Analyzer(packageRoot as AbsolutePath);
+  const analysis = analyzer.analyzePackage();
+  await Promise.all(
+    (frameworks as FrameworkName[])
+      .map((framework) => {
+        const importer = frameworkGenerators[framework];
+        if (importer === undefined) {
+          throw new Error(`No generator exists for framework '${framework}'`);
```

Comment:
Do you want a synchronous error here?

Reply:
Pulled validation out to be sync.

Reply:
Oh, it already was synchronous since it was building the array for Promise.all(), but I didn't know the intention, since it was next to an async map, which would have async errors.